File: internal/analyze/java/testdata/conditions.java
package com.acme.app;

class Conditions {
    boolean both(boolean a, boolean b) {
        return a && b;                     // condition 2
    }

    boolean either(boolean a, boolean b, boolean c) {
        return a && b || c;                // condition 3
    }

    boolean negated(boolean a, boolean b, boolean x) {
        return !(a || b) && x;             // condition 3 — flattened through ! and ( )
    }

    boolean compare(int x) {
        return x > 1;                      // condition 0 — a comparison is not a clause
    }

    int bits(int a, int b) {
        return a & b | 3;                  // condition 0 — bitwise operators are not clauses
    }
}

File: internal/analyze/golang/testdata/conditions.go
// conditions.go — condition clauses

package app

type Conditions struct{}

func (Conditions) Both(a, b bool) bool {
	return a && b // condition 2
}

func (Conditions) Either(a, b, c bool) bool {
	return a && b || c // condition 3
}

func (Conditions) Negated(a, b, x bool) bool {
	return !(a || b) && x // condition 3 — flattened through ! and ( )
}

func (Conditions) Compare(x int) bool {
	return x > 1 // condition 0 — a comparison is not a clause
}

func (Conditions) Bits(a, b int) int {
	return a&b | 3 // condition 0 — bitwise operators are not clauses
}

// unit Conditions: struct, condition 8, code_branch 0
